### ISSUE TITLE
Hall of Blame: The upgrade wracked my PC (under following conditions)

### ISSUE THREAD

**Issue by th-wilde** [th-wilde/veracrypt-w10-patcher#3]
Hello,
this issue is created to track configurations in witch a upgrade failed. If the upgrade failed for you, please share your disappointment here with some details. This will help to find the cause of the failure and prevent other users from doing the same mistakes.

Please use the following template for your comment to keep some structure. If you do not know some of the requested details, just let them empty.

```markdown
Damn! Wired things like ..... happened and now my PC is burning. 

This was my configuration:
**Upgrade Method:** [ReflectDrivers or MediaPatcher]
**Windows Edition:** [Pro ,Home, Education, Enterprise, ...]
**Architecture:** [64Bit, 32Bit]
**Firmeware/-Mode:** [BIOS, (U)EFI] - If your (U)EFI uses the legacy-/CSM-mode (https://en.wikipedia.org/wiki/Unified_Extensible_Firmware_Interface#CSM) to start Windows choose *BIOS*
**Encryption-Mode:**  [entire system drive, windows partition]
**Upgrade from Version:** [1507, 1511, 1607, 1703, ...]
**Upgrade to Version:** [1511, 1607, 1703, 1709, ...]
**Media used to Upgrade:** [media creation tool, name of your source like MSDN, MSDNAA/DreamSpark/Imagine/OnTheHub, ...]
**Installed VeraCrypt-Version:** [1.19, 1.20, 1.21, 1.22, 1.23, ...]
```

**Comment by adiiik** [comment]
Hello. I had run your script placed in USB drive created by media creation tool, and I have run setup.exe. Unfortunately, my Windows have started normally after rebooting.

This was my configuration:
**Windows Edition:** [Pro]
**Architecture:** [64Bit]
**Firmeware/-Mode:** [BIOS]
**Encryption-Mode:**  [windows partition]
**Upgrade from Version:** [1703]
**Upgrade to Version:** [1709]
**Media used to Upgrade:** [media creation tool]
**Installed VeraCrypt-Version:** [1.21]

**Comment by th-wilde** [comment]
@adiiik I don't now what went wrong for you. Maybe my explanation in Issue #5 helps you to get the Upgrade working.

**Comment by adiiik** [comment]
@th-wilde I thought that I don't have UEFI, but after rebooting to BIOS I turned out that I have. I will try once again in a few days.

**Comment by adiiik** [comment]
@th-wilde  I had tried to run bcdedit command, but I got an error: An error occured while attempting to reference the specified entry.

**Comment by SandyShores** [comment]
Hey, I got a very similar if not the same error when trying to run bcdedit as admin - I'm guessing we're missing something obvious but I have no idea what. I'm trying to upgrade from 10 x64 1607 14393.1770 to 1703, later to 1709.

Thanks for any hints (and the patcher of course) :)

EDIT: Well, according to this [https://www.novell.com/support/kb/doc.php?id=7009310](https://www.novell.com/support/kb/doc.php?id=7009310) the default Windows boot manager entry is missing. It can be added by running "bcdedit /default {current}" but I have no idea if that will ruin booting via VeraCrypt... I'm afraid it might so I'd rather hear the opinion of someone who knows what the heck he's doing before I try that.

Output of running bcdedit.exe on the currently installed Win 10 (took out the UIDs):

Windows-Start-Manager
Bezeichner              {bootmgr}
device                  partition=\Device\HarddiskVolume8
description             Windows Boot Manager
locale                  de-DE
inherit                 {globalsettings}
default                 {current}
resumeobject            {uid-A}
displayorder            {current}
toolsdisplayorder       {memdiag}
timeout                 30

Windows-Startladeprogramm
Bezeichner              {current}
device                  partition=C:
path                    \Windows\system32\winload.exe
description             Windows 10
locale                  de-DE
inherit                 {bootloadersettings}
recoverysequence        {uid-B}
recoveryenabled         Yes
allowedinmemorysettings 0x15000075
osdevice                partition=C:
systemroot              \Windows
resumeobject            {uid-A}
nx                      OptIn
bootmenupolicy          Standard
hypervisorlaunchtype    Off

**Comment by SandyShores** [comment]
Well thanks for not pointing out my stupidity. The problem is that the Windows install was NOT in fact done in UEFI mode but legacy, as can be seen by running msinfo32 and looking at the BIOS mode.
Confusingly my BIOS/UEFI setting (at system boot i mean) for "Launch Storage OpROM Policy" is on "UEFI only" so uh - yeah. Guess that means something else than I figured.

I got a different error on my first try installing the patched 1703, so I assumed it was because I had installed Win in UEFI mode. I'll get back with the actual error message in a few ;)

EDIT: Umm - I still have no idea what went wrong the first time with 1703 but the 1709 upgrade worked just fine on the first try after patching it with your tool. Thanks a lot!

**Comment by JamesLDoty** [comment]
Sony VAIO notebook, about 4 years old:
Windows Edition: [Pro]
Architecture: [64Bit]
Firmeware/-Mode: [BIOS]
Encryption-Mode: [windows partition]
Upgrade from Version: [1703]
Upgrade to Version: [1709]
Media used to Upgrade: [media creation tool]
Installed VeraCrypt-Version: [1.21]

I first tried the script on a USB drive: seemed to work okay.  Rebooted, and double clicked on setup.exe.  The small rectangular logo appeared in the center of the screen and it hung.  I let it go for more than a half hour, nothing.

Reran the media creation tool and created an ISO file.  Used 7Zip to extract ISO files to a directory, ran your script and everything seemed to go fine.  Ran setup.exe and everything seemed to go fine, but when came time to do the installation, it wanted to do a full install of Windows 10, though I did have the option of saving all my Files and Apps.  So I went ahead.  It went through the whole process, and on the first reboot it aborted and reverted to 1703.

**Comment by Dalamar6** [comment]
**Windows Edition:** [Enterprise]
**Architecture:** [64Bit]
**Firmware/-Mode:** [(U)EFI] 
*BIOS* latest for G752vm
**Encryption-Mode:**  [entire system drive]
**Upgrade from Version:** [1607]
**Upgrade to Version:** [1709]
**Media used to Upgrade:** [usb, after this script]
**Installed VeraCrypt-Version:** [1.21]

Typical SAFE_OS fail.
I did bcdedit /set {fwbootmgr} displayorder {bootmgr} /addlast from WINDOWS as there should be no reason to need to boot installation media.

Something might have been messed up from BIOS updates clearing CMOS as Veracrypt rescue disc couldn't fully restore uefi features(I had to enable F8 safe boot menu even).. UEFI sucks whale dong, MS should be sued.
Or it could be the terrible UEFI implementation ASSUS use. I will probably try MSI workstation later.

**Comment by purpleducks** [comment]
After running the patcher script under admin privileges, I tried running the setup.exe file under admin privileges, but it showed an error message saying "Failed to launch the setup application". I've tried starting from scratch and patching it again, but that didn't work either. Any ideas?

This was my configuration:
**Windows Edition:** Home
**Architecture:** 64Bit
**Firmeware/-Mode:** (U)EFI
**Encryption-Mode:**  windows partition
**Upgrade from Version:** 1703
**Upgrade to Version:** 1709
**Media used to Upgrade:** media creation tool - Windows USB Media creator
**Installed VeraCrypt-Version:** 1.20-BETA

**Comment by TCSFrank** [comment]
**Windows Edition:** [Home]
**Architecture:** [64bit]
**Firmeware/-Mode:** [BIOS]
**Encryption-Mode:**  [Entire System Drive]
**Upgrade from Version:** [1703]
**Upgrade to Version:** [1709]
**Media used to Upgrade:** [Rufus, USB]
**Installed VeraCrypt-Version:** [1.21]

SAFE_OS error after initial attempt. bcdedit errored while attempting to reference the specified entry. patch also had an error about incorrect drive, but I ignored it as per the instructions.

**Comment by FrustratedBurrito** [comment]
Hello.
I booted into the updated usb drive and ran the command prompt command.

It fails with the following:
The requested system device cannot be identified due to multiple indistinguishable devices potentially matching the identification criteria.

Please help!

This was my configuration:
**Windows Edition:** [Pro]
**Architecture:** [64Bit]
**Firmeware/-Mode:** [(UEFI]
**Encryption-Mode:**  [windows partition] (I'm pretty sure I left it at default in VC when encrypting)
**Upgrade from Version:** [1703]
**Upgrade to Version:** [1709]
**Media used to Upgrade:** [media creation tool USB]
**Installed VeraCrypt-Version:** [1.21]

**Comment by FrustratedBurrito** [comment]
I have simply unplugged my second drive before booting to solve my problem. I typed the listed bcdedit command in.
Now I get the following:
The boot configuration data store could not be opened.
The requested system device could not be found.

**Comment by Naudiz** [comment]
Failed 2 times yesterday. I tried to install from USB and afterwards from (encrypted) C: drive. Behavior is just the same as with the regular Windows-Update-function: After first reboot, it takes me to the blue screen (not bluescreen, lol) where I have to choose my keyboard-layout and afterwards see the tiles for "start windows", "shutdown", "advanced options", etc.
After another reboot, I'm in 1703 again and it greets me with the SAFE_OS error.

bcdedit commands always always report success, but it's weird that they don't seem to have any effect. When running bcdedit without parameters, I see the overview of entries. After I enter the /addlast command, it's the same as before. When I enter the /remove command, it's still the same.
So basically, I can run the /addlast or /remove command unlimited times and they never change anything, but always report "success".
I guess the problem lies there somewhere, but I don't know what's wrong exactly and why bcdedit doesn't do anything.

I guess it's decrypt-update-encrypt for me -.-

This was my configuration:
**Windows Edition:** [Pro]
**Architecture:** [64Bit]
**Firmeware/-Mode:** [UEFI]
**Encryption-Mode:**  [windows partition]
**Upgrade from Version:** [1703]
**Upgrade to Version:** [1709]
**Media used to Upgrade:** [media creation tool; downloaded & unpacked ISO; tried installation both from USB and C:-drive]
**Installed VeraCrypt-Version:** [1.21]

**Comment by borgqueenx** [comment]
Error when executing the UEFI command line in the bootable drive:
The boot configuration data store could not be opened.
The requested system device could not be found.

EDIT: Fixed. Make sure in your bios settings, that you're booting from the USB in UEFI as well. My bios has 2 boot options for the same usb drive. Normal was selected, forcing it to UEFI did the trick.

This was my configuration:
Windows Edition: [Pro]
Architecture: [64Bit]
Firmeware/-Mode: [UEFI]
Encryption-Mode: [windows partition]
Upgrade from Version: [1607]
Upgrade to Version: [1709]
Media used to Upgrade: [media creation tool; downloaded & unpacked ISO, run script over files, repacked, put into bootable media with rufus and run bootable disk.
Installed VeraCrypt-Version: [1.21]

**Comment by David00** [comment]
I'm having two problems. First, when I run the bcdedit command from the bootable win10 tool, it fails with the message: "The system cannot find the specified file".

Second, when I try to execute the process, it appears to be working until the first reboot.  Upon the reboot, I get the screen that asks me what keyboard layout I want to use, followed by the Windows 10 troubleshooting screen.  I click exit and continue to Windows, and I'm back in my original environment with a window that comes up that says "We couldn't install Windows 10" and references error 0xC1900101 - 0x20017 - the installation failed in the SAFE_OS phase with an error during BOOT operation.

This was my configuration:
Windows Edition: Home
Architecture: 64 Bit
Firmeware/-Mode: UEFI
Encryption-Mode: System-Partition (this is what veracrypt is showing, although I thought I had done the entire drive)
Upgrade from Version: 1709
Upgrade to Version: 1803
Media used to Upgrade: Official media creation tool. Created both an SD card and a USB flash drive using the non-ISO method in the tool.  Both appeared to be patched successfully in the script.
Installed VeraCrypt-Version: [1.21]

I really don't want to decrypt my drive to upgrade, but it appears the built in L2TP VPN functionality is broken while my laptop is in between upgrades, and I need that functionality back to continue working efficiently.

**Comment by zwarfox** [comment]
Hello,

Tried upgrading today. 

Attempt 1 - Standard upgrade method - setup.exe
---
The upgrade failed, as expected, on the first reboot; the computer started up in WinRE instead of booting to Windows. Rebooted and the installer rolled Windows back to 1709. Error message from installer about "SAFE_OS" after starting Windows.

Attempt 2 - setup.exe with /ReflectDrivers and /PostOOBE switches
---
Command:
`setup.exe /ReflectDrivers "C:\Program Files\VeraCrypt" /PostOOBE "C:\ProgramData\VeraCrypt\SetupComplete.cmd"`

Setup failed immediately with an unspecified error. Both folders and SetupComplete.cmd exists.

Attempt 3 - MediaPatcher
---
Patched the installation media using the batch file. This time the upgrade got past the first and second reboot, but on the third displayed a BSOD (**DRIVER_IRQL_NOT_LESS_OR_EQUAL - storport.sys**). The installer tried to recover, but failed after a few extra reboots and the same BSOD. Error message from installer about "SAFE_OS" after starting Windows.

Attempt 4 - Decrypted system drive + standard upgrade method
---

Decrypted the system drive. Started the upgrade by running setup.exe from the mounted Windows 10 1809 ISO-file.

Experienced the same BSOD concerning storport.sys on the third reboot, but the installer was able to recover and the upgrade was successful.


This was my configuration:
---
**Upgrade Method**: manual, ReflectDrivers, MediaPatcher
**Windows Edition**: Pro
**Architecture**: 64Bit
**Motherboard**: ASUS Prime X370-Pro
**Firmware/-Mode**: BIOS
**Encryption-Mode**: entire system drive
**Upgrade from Version**: 1709
**Upgrade to Version**: 1809
**Media used to Upgrade**: ISO generated by Media Creation Tool
**Installed VeraCrypt-Version**: 1.23

**Comment by TinchoteSK** [comment]
Interesting situation.  On Truecrypt, with MediaPatcher. The update did go through. But upon login, Truecrypt would not open claiming an incorrect version of the driver was installed. It turns out that for some reason the patch (that I prepared on a different computer, but with the same version of TC) installed the 7.0 driver instead of the 7.1. TC would not load, nor would its installer/uninstaller. After some magic (replacing the .sys file against Windows' wishes) I was able to run TC and everything was normal. 

Here my configuration:
Upgrade Method: [MediaPatcher]
Windows Edition: [Pro]
Architecture: [64Bit]
Firmeware/-Mode: [BIOS]
Encryption-Mode: [entire system drive]
Upgrade from Version: [1809]
Upgrade to Version: [1903]
Media used to Upgrade: [media creation tool, files copied to another (encrypted) hd]
Installed VeraCrypt-Version: [TrueCrypt 7.1a]
